test/k8sT/Services.go

```diff
@@ -152,6 +163,85 @@ var _ = Describe("K8sServicesTest", func() {
 				service.ExpectContains(clusterIP, "ClusterIP is not present in the cilium service list")
 			}
 		}, 300)
+
+		It("Checks connectivity is correct during deployment scale", func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			type BackgroundTestAsserts struct {
+				res  *helpers.CmdRes
+				time time.Time
+			}
+			backgroundChecks := []*BackgroundTestAsserts{}
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				for {
+					select {
+					default:
+						res := kubectl.ExecPodCmd(
+							helpers.DefaultNamespace, appPods[helpers.App2],
+							helpers.CurlFail("http://%s/public", clusterIP))
+						assert := &BackgroundTestAsserts{
+							res:  res,
+							time: time.Now(),
+						}
+						backgroundChecks = append(backgroundChecks, assert)
+					case <-ctx.Done():
+						wg.Done()
+						return
+					}
+				}
+			}()
+			// Sleep a bit to make sure that the goroutine starts.
+			time.Sleep(50 * time.Millisecond)
+
```

Comment:
Up to this line I have no knowledge of what the test is supposed to do, please add comments on why the go routine was created